### PR TITLE
Move macauff import pipeline to macauff repo

### PR DESCRIPTION
**Pull request by delucchi-cmu**
This is mostly a copy/paste of the macauff import pipeline from https://github.com/astronomy-commons/hipscat-import/pull/186

This adds a hipscatted version of the `catalog_a`/`catalog_b` test data that was introduced for other unit tests, as well as a reminder of how to create those catalogs from the macauff-generated catalogs.

### CODE REVIEW

File: tests/lsdb_macauff/import_pipeline/create_catalogs.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+"""Script to convert generated test catalogs into hipscatted catalogs.
+
+See also lsdb_mc_data.py, for generating some data using macauff utils."""
+
+import hipscat_import.pipeline as runner
+from hipscat_import.catalog.arguments import ImportArguments
+
+
+def create_catalogs():
```
Comment (camposandro) [inline tests/lsdb_macauff/import_pipeline/create_catalogs.py:9]: Did you mean to leave the `input_file_list` and `output_path` pointing to your local directories?

File: tests/lsdb_macauff/import_pipeline/test_run_import.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+@pytest.mark.dask
+def test_object_to_object(
+    import_catalog_a,
+    import_catalog_b,
+    tmp_path,
+    import_metadata_yaml,
+    import_match_csv,
+    dask_client,
+):
+    """Test that we can create a MacauffArguments instance with two valid catalogs."""
```
Comment (camposandro) [inline tests/lsdb_macauff/import_pipeline/test_run_import.py:36]: Should this docstring be something along the lines of:
> Test that we can create an association catalog by running the Macauff import pipeline

File: tests/lsdb_macauff/import_pipeline/test_arguments.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            left_ra_column="ra",
+            left_dec_column="dec",
+            left_id_column="id",
+            right_catalog_dir=catwise_dir,
+            right_ra_column="source_ra",
+            right_dec_column="source_dec",
+            right_id_column="source_id",
+            input_path=test_data_dir,
+            input_format="csv",
+            metadata_file_path="ceci_n_est_pas_un_fichier.xml",
```
Comment (camposandro) [inline tests/lsdb_macauff/import_pipeline/test_arguments.py:166]: Just loved these French easter eggs 😄

**Review by camposandro — APPROVED** [review]

File: tests/lsdb_macauff/import_pipeline/test_run_import.py (hunk 2 above)
Comment (delucchi-cmu) [reply tests/lsdb_macauff/import_pipeline/test_run_import.py:36]: Yes - and feel free to use "suggestions" on PRs, particularly for tweaking wording here and there.

**Review by camposandro — APPROVED** [review]

**Pull request merged**